test/components/ai-dialog.spec.js

```diff
@@ -0,0 +1,54 @@
+import '../../test_bundle/main.test.bundle.js';
+
+let component;
+
+const click = (element, event) => new Promise((resolve) => {
+  const listener = () => {
+    component.removeEventListener(event, listener);
+    resolve('Clicked!');
+  };
+
+  component.addEventListener(event, listener);
+  element.dispatchEvent(new MouseEvent('click', { bubbles: true }));
+});
+
+describe('ai-dialog', () => {
+  beforeEach(() => {
+    component = document.createElement('ai-dialog');
+    document.documentElement.appendChild(component);
+  });
+
+  it('dialog should be rendered in the DOM', () => {
+    expect(document.querySelector('ai-dialog')).to.equal(component);
+  });
+
+  it('should open dialog on button press', (done) => {
+    const listener = () => {
+      component.removeEventListener('validated', listener);
```

Comment:
put assignments first
```suggestion
      const openButton = component.shadowRoot.querySelector('#open-button');
      const dialog = component.shadowRoot.querySelector('.ai-dialog');
      
      component.removeEventListener('validated', listener);
```